table/block_based/block_based_table_builder.cc

```diff
@@ -644,11 +733,93 @@ struct BlockBasedTableBuilder::ParallelCompressionRep {
       block_rep_buf[i].keys.reset(new Keys());
       block_rep_buf[i].slot.reset(new BlockRepSlot());
       block_rep_buf[i].status = Status::OK();
+      block_rep_buf[i].is_first_block = false;
       block_rep_pool.push(&block_rep_buf[i]);
     }
   }
 
   ~ParallelCompressionRep() { block_rep_pool.finish(); }
+
+  // Make a block prepared to be emitted to compression thread
+  // Used in non-buffered mode
+  BlockRep* PrepareBlock(CompressionType compression_type,
+                         const Slice* first_key_in_next_block,
+                         BlockBuilder* data_block) {
+    BlockRep* block_rep =
+        PrepareBlockInternal(compression_type, first_key_in_next_block);
```

Comment:
Seems there's still concurrent read/write issue:
> There may be a Lock Consistency Violation: Read/Write race. Non-private method `rocksdb::BlockBasedTableBuilder::ParallelCompressionRep::EmitBlock` reads without synchronization from `this->is_waiting_first_block`. Potentially races with write in method `rocksdb::BlockBasedTableBuilder::ParallelCompressionRep::EmitBlock.`

Reply:
Replaced it with atomic variable.